presto-orc/src/main/java/com/facebook/presto/orc/writer/SliceDictionaryColumnWriter.java

```diff
@@ -362,6 +401,26 @@ private void bufferOutputData()
         presentStream.close();
     }
 
+    private LongInputStream getTempDictionaryIdInputStream()
```

Comment:
I'm guessing you considered this, but would it be reasonable to have `LongOutputStream` have a `isLongInputStream()`?